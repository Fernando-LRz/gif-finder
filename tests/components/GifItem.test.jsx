import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {
    
    const title = 'spider-man fighting gif';
    const url = 'https://gif.com/spider-man-fighting.jpg'

    test('Hacer match con el snapshot', () => {

        const { container } = render( <GifItem title={ title } url={ url } /> ); 
        expect( container ).toMatchSnapshot();

    });

    test('Mostrar la imágen con la URL y el ALT establecido', () => {
        
        render( <GifItem title={ title } url={ url } /> );
        // screen.debug();

        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );

        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe( url );
        expect( alt ).toBe( title );

    });

    test('Mostrar el título en el componente', () => {
        
        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();

    });

});