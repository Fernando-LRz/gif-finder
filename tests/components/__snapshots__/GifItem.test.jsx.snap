// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="spider-man fighting gif"
      src="https://gif.com/spider-man-fighting.jpg"
    />
    <p>
      spider-man fighting gif
    </p>
  </div>
</div>
`;
